core/src/main/java/uk/gov/justice/services/core/handler/HandlerMethodInvoker.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.justice.services.core.handler;
+
+
+import static java.lang.Class.forName;
+
+import uk.gov.justice.services.common.converter.jackson.ObjectMapperProducer;
+import uk.gov.justice.services.messaging.Envelope;
+import uk.gov.justice.services.messaging.JsonEnvelope;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+import javax.json.JsonObject;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class HandlerMethodInvoker {
+
+    public Object invoke(final Object handlerInstance,
+                         final Method handlerMethod,
+                         final JsonEnvelope jsonEnvelope)
+            throws InvocationTargetException, IllegalAccessException, ClassNotFoundException, IOException {
+
+        final Class<?> parameterClass = handlerMethod.getParameterTypes()[0];
+        if(parameterClass == JsonEnvelope.class) {
```

Comment:
Instead of "==" maybe JsonEnvelope.class.equals(parameterClass) would be better

Reply:
As long as you are in the same virtual machine then equals(...) should be equivalent to '=='

Reply:
Yes you are right...I was thinking of something else...scratch my comment